### ISSUE TITLE
Fatal failure at src/theory/rewriter.cpp:348

### ISSUE THREAD

**Issue by muchang** [cvc5/cvc5#5605]
```
[517] % cvc4 -q small.smt2
Fatal failure within CVC4::Node CVC4::theory::Rewriter::rewriteTo(CVC4::theory::TheoryId, CVC4::Node, CVC4::TConvProofGenerator*) at src/theory/rewriter.cpp:348
Check failure
 r2.d_node == response.d_node
...
Aborted
[518] % 
[518] % cat small.smt2
(declare-fun a () (Array (_ BitVec 32) (_ BitVec 8)))
(assert
(let ((b (concat (select a (_ bv2 32)) (concat (select a (_ bv1 32)) (select a (_ bv0 32))))))
 (let ((c (concat (select a (_ bv5 32)) (concat (select a (_ bv4 32)) (concat (select a (_ bv3 32)) b)))))
 (let ((d ((_ to_fp 11 53) (concat (select a (_ bv7 32)) (concat (select a (_ bv6 32)) c)))))
  (let ((e (fp.add roundNearestTiesToEven (fp.abs d) ((_ to_fp 11 53) (_ bv5781251510991923850 64)))))
  (let ((f (fp.mul roundNearestTiesToEven e e)))
   (not (fp.geq (fp.sub roundNearestTiesToEven f f) ((_ to_fp 11 53) (_ bv9218868437227405312 64))))))))))
(check-sat)
[519] %
```
This one doesn't depend on bvcomp.

Commit: a3395c1

**Comment by nafur** [comment]
```
(declare-fun b () (_ FloatingPoint 8 24))
(assert (= (fp.sub roundNearestTiesToEven (fp.mul roundNearestTiesToEven b b)
  (fp.mul roundNearestTiesToEven b b)) (fp (_ bv1 1) (_ bv1 8) (_ bv1 23))))
(check-sat)
```

**Comment by zhendongsu** [comment]
Another similar and related test:
```
[694] % cvc5 -q small.smt2 
Fatal failure within cvc5::Node cvc5::theory::Rewriter::rewriteTo(cvc5::theory::TheoryId, cvc5::Node, cvc5::TConvProofGenerator*) at /local/suz-local/software/CVC4/src/theory/rewriter.cpp:352
Check failure

 r2.d_node == response.d_node
(bvor (NAN a) (bvand (ZERO a) (INF a))) != (bvnot (bvnot (bvor (NAN a) (bvand (ZERO a) (INF a)))))
Aborted
[695] % 
[695] % cat small.smt2 
(declare-fun a () Float32)
(assert (fp.isZero (fp.sub RNE (fp.mul RNE a a) (fp.mul RNE a a))))
(check-sat)
[696] % 
```

**Comment by 4tXJ7f** [comment]
Thanks for the report and the examples. I have been able to confirm that all the benchmarks have been fixed in #6624.
